includes/internal/quiz-submission/answer/models/class-comments-based-answer.php

```diff
@@ -0,0 +1,23 @@
+<?php
+/**
+ * File containing the Comments_Based_Answer class.
+ *
+ * @package sensei
+ */
+
+namespace Sensei\Internal\Quiz_Submission\Answer\Models;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Class Comments_Based_Answer.
+ *
+ * @internal
+ *
+ * @since $$next_version$$
+ */
+class Comments_Based_Answer extends Answer_Abstract {
+
```

Comment:
As we don't have an id for this version, what do you think about throwing an exception here?
Also we don't need it in the constructor.

Reply:
Makes sense. I've updated it in dc2347f69f2caf262149ad4b1d3ee83f7637383b.

I've also updated (https://github.com/Automattic/sensei/pull/7178/commits/73889e3bb8a4da157768063741bc7cae2ff2f162) the `Grade_Repository->create` method to accept the answer model instance instead of an id. This change fixes the issue of having to check the answer model type and having to pass an ID of 0 if it's a comment type.

Reply:
Thanks @m1r0! Feels better :)